rand_core/src/lib.rs

```diff
@@ -208,6 +208,20 @@ pub trait RngCore {
 /// [`BlockRngCore`]: block::BlockRngCore
 pub trait CryptoRng {}
 
+/// An extension trait to support trait objects that implement [`RngCore`] and
+/// [`CryptoRng`]. Upcasting to [`RngCore`] is supported via the
+/// [`CryptoRngCore::upcast`] method.
+pub trait CryptoRngCore: RngCore {
+    /// Upcast to an [`RngCore`] trait object.
+    fn upcast(&mut self) -> &mut dyn RngCore;
```

Comment:
This should perhaps be called `as_rngcore`? (An `_mut` suffix is not necessary since a non-mut RNG is useless.)

Reply:
I agree, your suggestion follows the Rust naming conventions more closely.